shape-detection/endtoend-barcodedetection.https.html

```diff
@@ -0,0 +1,399 @@
+
+const videoTests = {
+    "barcodes.mov": [
+        {time: 0.5, test: aztecCorrection},
+        {time: 1.5, test: aztecFull},
+        {time: 2.5, test: aztecLayers},
+        {time: 3.5, test: aztec},
+        {time: 4.5, test: code128Height},
+        {time: 5.5, test: code128},
+        {time: 6.5, test: pdf417Columns},
+        {time: 7.5, test: pdf417Compact},
+        {time: 8.5, test: pdf417Compaction},
+        {time: 9.5, test: pdf417Correction},
+        {time: 10.5, test: pdf417Rows},
+        {time: 11.5, test: pdf417Square},
+        {time: 12.5, test: pdf417Taller},
+        {time: 13.5, test: pdf417Wider},
+        {time: 14.5, test: pdf417},
+        {time: 15.5, test: qrBottomLeft},
+        {time: 16.5, test: qrBottomRight},
+        {time: 17.5, test: qrCenter},
+        {time: 18.5, test: qrH},
+        {time: 19.5, test: qrL},
+        {time: 20.5, test: qrM},
+        {time: 21.5, test: qrQ},
+        {time: 22.5, test: qrTopLeft},
+        {time: 23.5, test: qrTopRight}]};
+
+// All the fields in FaceDetectorOptions are hints, so they can't be tested.
+const barcodeDetector = new BarcodeDetector();
+
+async function testImage(imageBitmapSource, test) {
+    const supportedFormats = await BarcodeDetector.getSupportedFormats();
+    if (!supportedFormats.includes(test.format))
+        return;
+    const detectedBarcodes = await barcodeDetector.detect(imageBitmapSource);
+    assert_equals(detectedBarcodes.length, 1);
+    const detectedBarcode = detectedBarcodes[0];
+    checkBoundingBox(detectedBarcode.boundingBox, test.barcode.boundingBox, test.barcode.fuzziness);
+    assert_equals(detectedBarcode.rawValue, test.payload);
+    assert_equals(detectedBarcode.format, test.format);
+    assert_equals(detectedBarcode.cornerPoints.length, 4);
+    const [topLeft, topRight, bottomRight, bottomLeft] = detectedBarcode.cornerPoints;
+    assert_true(pointIsNear(topLeft, test.topLeft.position, test.topLeft.fuzzinessX, test.topLeft.fuzzinessY));
+    assert_true(pointIsNear(topRight, test.topRight.position, test.topRight.fuzzinessX, test.topRight.fuzzinessY));
+    assert_true(pointIsNear(bottomRight, test.bottomRight.position, test.bottomRight.fuzzinessX, test.bottomRight.fuzzinessY));
+    assert_true(pointIsNear(bottomLeft, test.bottomLeft.position, test.bottomLeft.fuzzinessX, test.bottomLeft.fuzzinessY));
+}
+
+promise_test(async t => {
+    for (imageTest of imageTests) {
+        const imageElement = document.createElement("img");
+        imageElement.src = `resources/${imageTest.name}`;
+        await imageLoadedPromise(imageElement);
+        assert_equals(imageElement.complete, true);
+        await testImage(imageElement, imageTest);
+    }
+}, "HTMLImageElement");
+
+// Intentionally don't test SVGImageElement. The spec https://html.spec.whatwg.org/multipage/canvas.html#canvasimagesource says it's supposed to be
```

Comment:
Nit: line length.

Reply:
Is there a policy about line length in WPT?

Reply:
Even if there is, I'd like to practice some civil disobedience and intentionally not obey silly rules like this